lib/config.js

```diff
@@ -195,7 +195,7 @@ const getFullConfig = async ({
     isLatest,
     // whether to install and update npm in ci
     // only do this if we aren't using a custom path to bin
-    updateNpm: !npmPath.isLocal,
+    updateNpm: !npmPath.isLocal && pkgConfig.updateNpm,
```

Comment:
`updateNpm` controls whether we run `npm install -g npm@latest` in CI.

Previously this was controlled by whether the path to npm was local, which was only used by the CLI since the CLI repo should always use itself as `npm` when in CI.

This commit adds a config option `updateNpm` which defaults to `true`, but can be set to false to override this behavior. This is useful when testing in very old Node versions, where we are ok using whatever `npm` is packaged with it.